src/uct/sm/mm/base/mm_ep.h

```diff
@@ -18,6 +18,13 @@ KHASH_INIT(uct_mm_remote_seg, uintptr_t, uct_mm_remote_seg_t, 1,
            kh_int64_hash_func, kh_int64_hash_equal)
 
 
+/* owner of segment process information. we have to cache this value
+ * because some transports terminate segment when process gone (xpmem) */
+typedef struct uct_mm_keepalive_info {
+    char       proc[UCT_SM_PROC_DIR_NAME_MAX_LEN]; /* Process owner proc dir */
+    ucs_time_t starttime;                          /* Process starttime */
```

Comment:
could do like this:
typedef struct uct_mm_keepalive_info {
 ucs_time_t start_time;
 char prof_file_path[0]
}
and then malloc according to actual size - and remove UCT_SM_PROC_DIR_NAME_MAX_LEN constant

Reply:
ok, done